Gems/Atom/Feature/Common/Code/Source/ReflectionProbe/ReflectionProbeFeatureProcessor.cpp

```diff
@@ -177,66 +179,214 @@ namespace AZ
             }
         }
 
-        ReflectionProbeHandle ReflectionProbeFeatureProcessor::AddProbe(const AZ::Transform& transform, bool useParallaxCorrection)
+        ReflectionProbeHandle ReflectionProbeFeatureProcessor::AddReflectionProbe(const AZ::Transform& transform, bool useParallaxCorrection)
         {
-            AZStd::shared_ptr<ReflectionProbe> reflectionProbe = AZStd::make_shared<ReflectionProbe>();
+            ReflectionProbePtr reflectionProbe = AZStd::make_shared<ReflectionProbe>();
             reflectionProbe->Init(GetParentScene(), &m_reflectionRenderData);
             reflectionProbe->SetTransform(transform);
             reflectionProbe->SetUseParallaxCorrection(useParallaxCorrection);
+
             m_reflectionProbes.push_back(reflectionProbe);
+            m_reflectionProbeMap[reflectionProbe->GetUuid()] = reflectionProbe;
+
             m_probeSortRequired = true;
 
-            return reflectionProbe;
+            return reflectionProbe->GetUuid();
         }
 
-        void ReflectionProbeFeatureProcessor::RemoveProbe(ReflectionProbeHandle& probe)
+        void ReflectionProbeFeatureProcessor::RemoveReflectionProbe(ReflectionProbeHandle& handle)
         {
-            AZ_Assert(probe.get(), "RemoveProbe called with an invalid handle");
+            if (!ValidateHandle(handle))
+            {
+                return;
+            }
 
-            auto itEntry = AZStd::find_if(m_reflectionProbes.begin(), m_reflectionProbes.end(), [&](AZStd::shared_ptr<ReflectionProbe> const& entry)
+            ReflectionProbePtr reflectionProbe = m_reflectionProbeMap[handle];
+            auto itEntry = AZStd::find_if(m_reflectionProbes.begin(), m_reflectionProbes.end(), [&](ReflectionProbePtr const& entry)
             {
-                return (entry == probe);
+                return (entry == reflectionProbe);
             });
 
             AZ_Assert(itEntry != m_reflectionProbes.end(), "RemoveProbe called with a probe that is not in the probe list");
+
             m_reflectionProbes.erase(itEntry);
+            m_reflectionProbeMap.erase(handle);
+        }
+
+        void ReflectionProbeFeatureProcessor::SetOuterExtents(const ReflectionProbeHandle& handle, const Vector3& outerExtents)
+        {
+            if (!ValidateHandle(handle))
+            {
+                return;
+            }
+
+            m_reflectionProbeMap[handle]->SetOuterExtents(outerExtents);
```

Comment:
In all these functions you are finding the handle twice, once for validation and then again to actually access the data. I expect we might have many probes in the scene, so consider changing ValidateHandle to be FindProbe, which does the validation but also returns an iterator.

Reply:
I considered this, but it's a constant-time operation so the cost should not be a major issue.